### ISSUE TITLE
error 0x8007010b when launching `wsl -d Ubuntu-18.04

### ISSUE THREAD

**Issue by Trid-collab** [microsoft/terminal#11245]
### Windows Terminal version (or Windows build number)

Windows Terminal Version: 1.10.2383.0

### Other Software

windows terminal + WSL +Ubuntu

### Steps to reproduce

I have wsl - Ubuntu as my default profile in WIndows Terminal.

I have noticed whenever I am trying to launch the terminal by default to the Ubuntu home directory  using the "wsl -d Ubuntu-18.04" in my Terminal Config file I am getting the following error and I am not able to open wsl.

[error 0x8007010b when launching `wsl -d Ubuntu-18.04']
Could not access starting directory "//wsl$/Ubuntu-18.04/home/tridib"

However if open the command prompt and type wsl I am able to launch Ubuntu which opens it by default on my windows user folder and then I need to cd into the Ubuntu home directory.

Is there something I am missing in my config. 

### Actual Behavior

![image](https://user-images.githubusercontent.com/61322708/133550244-13374a6d-71ff-4237-a6a1-26aa910e11c5.png)

**Comment by zadjii-msft** [comment]
Curious. Does this work right if you first open a conhost window running wsl (i.e. just run `wsl.exe`) _then_ open the Terminal? I'm wondering if we're trying to access the `wsl$` mount before it actually exists, because the WSL service hasn't started yet.

**Comment by Trid-collab** [comment]
If I run win+r and type wsl it opens the Ubuntu terminal. Then as long that is open I can use the Windows Terminal to open Ubuntu Terminal normally without any issues

**Comment by elsaco** [comment]
@Trid-collab this is how I start Ubuntu in my home directory:

```
            {
                "commandline": "wsl.exe ~ -d Ubuntu-20.04",
                "guid": "{07b52e3e-de2c-5db4-bd2d-ba144ed6c273}",
                "hidden": false,
                "name": "Ubuntu-20.04",
                "source": "Windows.Terminal.Wsl"
            },
            {
                "commandline": "wsl ~ -d Ubuntu-18.04",
                "guid": "{c6eaf9f4-32a7-5fdc-b5cf-066e8a4b1e40}",
                "hidden": false,
                "name": "Ubuntu-18.04",
                "source": "Windows.Terminal.Wsl"
            },
```

Something to get you running until "startingDirectory" is being fixed :)

**Comment by zadjii-msft** [comment]
@DHowett Shouldn't #9223 have fixed this?

🤦 doh, that's only in 1.11 preview. @Trid-collab Could you try installing "Windows Terminal (Preview)", and reboot the machine (to make sure the WSL service stops), then try this with the Preview version of the Terminal? I'm pretty sure this is fixed in 1.11.

**Comment by Trid-collab** [comment]
Thank you.  let me check. I do not have access to the preview ver right now.

But with the stable release I noticed that sometimes the stable version worked properly with out any errors and sometime it came up with the issues like the one mentioned . So I am confused

**Comment by zadjii-msft** [comment]
> But with the stable release I noticed that sometimes the stable version worked properly with out any errors and sometime it came up with the issues like the one mentioned . So I am confused

Yea, that's because the `//wsl$` path isn't accessible until the WSL service (lxss) is already running. So until that service starts, the Terminal thinks that any paths under that root aren't valid paths. 

I'm going to tentatively call this fixed by #9223, and when 1.11 moves to stable, if that hasn't fixed this issue for you, we can reopen then. Sound good?

**Comment by Trid-collab** [comment]
Thank you. WIll check it up with 1.11 when I have an access to it
